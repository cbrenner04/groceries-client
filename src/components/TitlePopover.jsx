import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Popover, Row } from 'react-bootstrap';

const TitlePopover = ({ message, title }) => (
  <OverlayTrigger
    trigger={['click']}
    placement="top"
    rootClose={true}
    overlay={
      <Popover>
        <Popover.Content data-test-id="popover-content">{message}</Popover.Content>
      </Popover>
    }
  >
    {({ ref, ...triggerHandler }) => (
      <Row className="title-popover-row">
        <h2 className="text-capitalize">{title}</h2>
        <i
          ref={ref}
          {...triggerHandler}
          className="far fa-question-circle float-right text-secondary"
          data-test-id={`${title}-popover`}
        />
      </Row>
    )}
  </OverlayTrigger>
);

TitlePopover.propTypes = {
  message: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

export default TitlePopover;
