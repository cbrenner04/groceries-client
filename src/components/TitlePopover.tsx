import React, { ReactNode } from 'react';
import { OverlayTrigger, Popover, Row } from 'react-bootstrap';

interface ITitlePopoverProps {
  message: ReactNode;
  title: string;
}

const TitlePopover: React.FC<ITitlePopoverProps> = ({ message, title }) => (
  <OverlayTrigger
    trigger={['click']}
    placement="top"
    rootClose={true}
    overlay={
      <Popover>
        <Popover.Body data-test-id="popover-content">{message}</Popover.Body>
      </Popover>
    }
  >
    {({ ref, ...triggerHandler }) => (
      <Row className="m-0">
        <h2 className="text-capitalize pe-0 w-auto">{title}</h2>
        <i
          ref={ref}
          {...triggerHandler}
          className="far fa-question-circle float-end text-secondary ps-0 w-auto"
          data-test-id={`${title}-popover`}
        />
      </Row>
    )}
  </OverlayTrigger>
);

export default TitlePopover;
