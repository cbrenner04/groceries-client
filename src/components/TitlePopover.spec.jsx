import React from 'react';
import { render } from '@testing-library/react';

import TitlePopover from './TitlePopover';

describe('TitlePopover', () => {
  it('renders', () => {
    const { container } = render(<TitlePopover title="foo" message="bar" />);

    expect(container).toMatchSnapshot();
  });
});
