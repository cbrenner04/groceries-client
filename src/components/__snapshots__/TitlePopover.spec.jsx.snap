// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitlePopover renders 1`] = `
<div>
  <div
    class="row"
    style="margin: 0px;"
  >
    <h2
      class="text-capitalize"
    >
      foo
    </h2>
    <i
      class="far fa-question-circle float-right text-secondary"
    />
  </div>
</div>
`;
